src/providerAsMiddleware.ts

```diff
@@ -9,21 +12,9 @@ import type {
 export function providerAsMiddleware(
   provider: SafeEventEmitterProvider,
 ): JsonRpcMiddleware<JsonRpcParams, Json> {
-  return (req, res, _next, end) => {
-    // send request to provider
-    provider.sendAsync(
-      req,
-      (err: unknown, providerRes: PendingJsonRpcResponse<any>) => {
-        // forward any error
-        if (err instanceof Error) {
-          return end(err);
-        }
-        // copy provider response onto original response
-        Object.assign(res, providerRes);
-        return end();
-      },
-    );
-  };
+  return createAsyncMiddleware(async (req, res) => {
+    res.result = await provider.request(req);
```

Comment:
Just today, @kanthesha and I re-released the EIP-1193 changes in `@metamask/eth-block-tracker` under [11.0.0](https://github.com/MetaMask/eth-block-tracker/releases/tag/v11.0.0) instead of 10.1.0, as we realized they were breaking. I think we will want to make sure that we publish this change under a major version and mark it as breaking as well, because consumers now need to use a provider that supports the `request` method, i.e., they will need to upgrade `@metamask/eth-json-rpc-provider` to upgrade to this change in this package.